libpod/image/pull.go

```diff
@@ -56,6 +58,17 @@ type pullStruct struct {
 }
 
 func (ir *Runtime) getPullStruct(srcRef types.ImageReference, destName string) (*pullStruct, error) {
+	img_part, err := decompose(destName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing destination name, wrong name format")
+	} else {
+		if !img_part.hasRegistry {
```

Comment:
Wait, is this safe? We use this for more than just load, right?

Reply:
from what I can tell it's just used in pull and in load. I don't know if it's safe, this was my attempt at doing what @baude advised me to do, and I fully accept I may have misinterpreted :)

Reply:
@baude Are we certain this is safe? This is reused code, and I don't want to go prepending `localhost` when we try and pull down, say, `busybox` or `alpine`.

Reply:
@mheon I definitely don't know all the cases, but for a pull of alpine or busybox, docker includes in the image their registry. As long as the other registries do the same, doing a podman pull <name in registry> should have the correct registry. Getting an image to have no registry attached was difficult enough I couldn't figure out a way to create a test without @edsantiago's tar

Reply:
(Assuming this code migrated into `getPullStruct`, the only user of `hasRegistry` in `pull.go`.)

No, this isn’t safe AFAICT.  `podman pull busybox` goes through `createNamesToPull` and is not affected — but the `getPullListFromRef` path is used for input like `podman pull docker://busybox`, and this change turns it into something like `skopeo copy docker://busybox containers-storage:localhost/busybox`.